beta-src/src/utils/state/gameApiSlice/extraReducers/fetchGameData/precomputeLegalOrders.ts

```diff
@@ -0,0 +1,524 @@
+import provincesMapData from "../../../../../data/map/ProvincesMapData";
+import TerritoryMap from "../../../../../data/map/variants/classic/TerritoryMap";
+import Province from "../../../../../enums/map/variants/classic/Province";
+import Territory from "../../../../../enums/map/variants/classic/Territory";
+import { UnitType } from "../../../../../models/enums";
+import { IProvinceStatus, ITerritory } from "../../../../../models/Interfaces";
+import { GameData } from "../../../../../state/interfaces/GameDataResponse";
+import GameOverviewResponse from "../../../../../state/interfaces/GameOverviewResponse";
+import { GameState } from "../../../../../state/interfaces/GameState";
+import GameStateMaps from "../../../../../state/interfaces/GameStateMaps";
+
+interface LegalVia {
+  dest: Territory;
+  provIDPaths: string[][];
+}
+
+interface LegalSupport {
+  src: Province;
+  dest: Province;
+  convoyProvIDPath: string[] | null;
+}
+
+interface LegalConvoy {
+  src: Territory;
+  dest: Territory;
+  // Concatenate these two arrays if you want the final convoy path.
+  // Do NOT modify these arrays.
+  provIDPath1: string[];
+  provIDPath2: string[];
+}
+
+export interface LegalOrders {
+  legalMoveDestsByUnitID: { [key: string]: Territory[] };
```

Comment:
It would be nice to put types here like `UnitID` and `ProvinceID`, even if these are just aliases for string.

```
type UnitID = string
```

Reply:
Good idea. Maybe better if we use type-enforcing accessor methods rather than raw access, but aliases would help with self documenting.